virtualizarr/manifests/manifest.py

```diff
@@ -54,6 +65,110 @@ def with_validation(
         return ChunkEntry(path=path, offset=offset, length=length)
 
 
+class ManifestStore(Store):
+    supports_writes: bool = False
+    supports_deletes: bool = False
+    supports_partial_writes: bool = False
+    supports_listing: bool = True
+
+    fs: AsyncFileSystem
+    vds: xr.Dataset
+
+    def __init__(self, fs, vds, read_only=True):
+        super().__init__(read_only=read_only)
+        self.fs = fs
+        self.vds = vds
+
+    async def clear(self) -> None:
+        self.fs = None
+        self.vds = None
+
+    def __str__(self) -> str:
+        return f"manifest://{id(self.vds)}"
+
+    def __repr__(self) -> str:
+        return f"ManifestStore('{self}')"
+
+    def __eq__(self, other: object) -> bool:
+        return (
+            isinstance(other, type(self))
+            and self.fs == other.fs
+            and xr.testing.assert_identical(self.vds, other.vds)
+        )
+
+    async def get(
+        self,
+        key: str,
+        prototype: BufferPrototype,
+        byte_range: None = None,  # could this optionally accept a RangeByteRequest?
+    ) -> Buffer | None:
+        if not self._is_open:
+            await self._open()
+        print("key: ", key)
+        print("key split: ", key.split("/"))
+        array_name, _, chunk_key = key.split("/")
+        # TODO: is this the best way?
+        url, offset, length = self.vds[array_name].data.manifest.dict()[chunk_key]
```

Comment:
I think this is a reason to improve the API of `ChunkManifest` to make it more dict-like (separately from this PR). Although I guess we already have the entire numpy array containing all references already in memory.